Modules/IO/ImageBase/include/itkIOTestHelper.h

```diff
@@ -75,55 +75,60 @@ class IOTestHelper
 
   template <typename ImageType, typename ImageIOType>
   static void
-  WriteImage(typename ImageType::Pointer & image,
+  WriteImage(typename ImageType::Pointer   image,
              const std::string &           filename,
              typename ImageIOType::Pointer imageio = nullptr)
   {
-
-    using WriterType = itk::ImageFileWriter<ImageType>;
-    auto writer = WriterType::New();
-
     if (imageio.IsNull())
     {
       imageio = ImageIOType::New();
     }
-
-    writer->SetImageIO(imageio);
-
-    writer->SetFileName(filename.c_str());
-
-    writer->SetInput(image);
-
-    try
-    {
-      writer->Update();
-    }
-    catch (const itk::ExceptionObject & err)
-    {
-      std::cerr << "Exception Object caught: " << std::endl << err << std::endl;
-      throw;
+    using WriterType = itk::ImageFileWriter<ImageType>;
+    { // Test valid filename writing
+      auto writer = WriterType::New();
+      writer->SetImageIO(imageio);
+      writer->SetFileName(filename);
+      writer->SetInput(image);
+      try
```

Comment:
Why not replace by `ITK_TRY_EXPECT_NO_EXCEPTION` from https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc04/Modules/Core/TestKernel/include/itkTestingMacros.h#L130?